### PR TITLE
added flexible voucher strategies [erc3525-flexible-voucher]

### PR DESCRIPTION
**Pull request by buchaoqun**
Fixes # .

Changes proposed in this pull request:
- added flexible voucher strategies
- 
-

### CODE REVIEW

**Comment by bonustrack** [comment]
@buchaoqun Can you add a readme for the strategy? Curious to know how it works, is it some kind of pay to vote system?

**Comment by buchaoqun** [comment]
> @buchaoqun Can you add a readme for the strategy? Curious to know how it works, is it some kind of pay to vote system?

Thank you for your attention to the our proposal based on ERC3525. Please allow me to explain in response to your question.
The strategy you see is based on ERC3525, a new asset standard. You can think of ERC3525 as an NFT, and Vesting Voucher is a new asset which encapsulated ERC20 tokens in the voucher.
Therefore, when users hold ERC3525, they are able to gain the corresponding governance power or weighs up the governance right as the time goes by, just like Curve's veCRV. 
This is not a pay to vote system.
The ERC3525 is mainly adopted by SolvProtocol and FuJiDAO, and there are more than 40 projects using ERC3525 (DoDo, Cls, HighStreat, USF, etc.) to lock their tokens.
So we need strategies like this to engage with our voucher holders. Thank you

**Comment by bonustrack** [comment]
Sound great thanks for the details!

File: src/strategies/erc3525-vesting-voucher/index.ts [hunk 1]
```diff
@@ -8,95 +8,90 @@ export const author = 'buchaoqun';
...
+    }
+    const walletIDToSnapshot: Record<string, Array<any>> =
+        await callCrucibleToSnapshot.execute();
 
-  const walletToWeights = {} as Record<string, number>;
-  for (const [walletID, snapshot] of Object.entries(walletIDToSnapshot)) {
-    const address = walletID.split('-')[0];
+    const walletToWeights = {} as Record<string, number>;
+    for (const [walletID, snapshot] of Object.entries(walletIDToSnapshot)) {
+        const address = walletID.split('-')[0];
```
Comment (ChaituVR) [inline src/strategies/erc3525-vesting-voucher/index.ts:83]: @buchaoqun Looks like it is messing up lint in this file

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/erc3525-vesting-voucher/index.ts [hunk 2]
```diff
@@ -8,95 +8,90 @@ export const author = 'buchaoqun';
...
-  addresses,
-  options,
-  snapshot
+    space,
+    network,
+    provider,
+    addresses,
+    options,
+    snapshot
 ) {
```
Comment (ChaituVR) [inline src/strategies/erc3525-vesting-voucher/index.ts:23]: @buchaoqun make sure it is not changing other strategies, revert changes from these files plz :)

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Comment by ChaituVR** [comment]
@buchaoqun I still see many changes in other strategy where the lint is messed up, try going here, https://github.com/snapshot-labs/snapshot-strategies/pull/740/files make sure it contains changes from your strategy

**Review by buchaoqun — COMMENTED** [review]
done

**Review by buchaoqun — COMMENTED** [review]
@ChaituVR done

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**